apps/snitch_core/test/data/model/package_item_test.exs

```diff
@@ -31,102 +29,113 @@ defmodule Snitch.Data.Model.PackageItemTest do
-         shipping_category_count: 1,
-         shipping_method_count: 1,
-         state_zone_count: 1
+           shipping_category_count: 1,
+           shipping_method_count: 1,
+           state_zone_count: 1
+    test "successful with valid params", context do
+      %{package_item: package_item} = new_package(context)
+      params = %{quantity: 95}
+      {:ok, updated_package_item} = PackageItem.update(params, package_item)
+      assert updated_package_item.id == package_item.id
+      assert updated_package_item.quantity != package_item.quantity
+      refute updated_package_item.backordered?
+    end
 
-    test "with valid params", context do
-      %{line_items: [line_item]} = context
+    test "with invalid params", context do
+      %{package_item: package_item} = new_package(context)
+      money = Money.new(-1, :USD)
+      updates = %{tax: money }
+      {:error, updated} = PackageItem.update(updates, package_item)
+      assert %{tax: ["must be equal or greater than 0"]} == errors_on(updated)
+    end
+  end
 
-      assert {:ok, package_item} =
-               PackageItem.create(%{
-                 @params
-                 | line_item_id: line_item.id,
-                   product_id: line_item.product_id,
-                   package_id: make_package(context).id
-               })
+  describe "delete/1" do
+    test "a package_item", context do
+      %{package_item: package_item} = new_package(context)
+      {:ok, _} = PackageItem.delete(package_item)
+      assert {:error, :package_item_not_found} == PackageItem.get(package_item.id)
+    end
+  end
 
-      assert package_item.backordered?
+  describe "get/1" do
+    test "with non-negative integer", context do
+      %{package_item: package_item} = new_package(context)
+      {:ok, new_package_item} = PackageItem.get(package_item.id)
+      assert new_package_item.id == package_item.id
+    end
 
-      assert {:ok, package_item} =
-               PackageItem.update(
-                 %{
-                   state: "destroyed",
-                   quantity: 5,
-                   delta: 0
-                 },
-                 package_item
-               )
+    test "with a map", context do
+      %{package_item: package_item} = new_package(context)
+      map = %{quantity: package_item.quantity, delta: package_item.delta}
+      {:ok, new_package_item} = PackageItem.get(map)
+      assert new_package_item.id == package_item.id
+    end
+  end
 
-      refute package_item.backordered?
+  describe "get_all/0" do
+    test "package_items", context do
+      %{package_item: package_item} = new_package(context)
+      assert PackageItem.get_all() != []
     end
   end
 
-  defp make_package(context) do
-    %{
-      order: order,
-      shipping_categories: [sc],
-      shipping_methods: [sm]
-    } = context
-
-    insert(
-      :package,
-      order: order,
-      shipping_category: sc,
-      shipping_method_id: sm.id
-    )
+
+  defp new_package(context) do
```

Comment:
Break the function into two one should create a `package` another should create `package_item` and you should pass params which are required explicitly. You can make use of the context. The way `embedded_shipping_methods` is passed.